pulp_smash/tests/pulp3/file/api_v3/test_sync.py

```diff
@@ -85,3 +86,70 @@ def do_test(self, download_policy, sync_mode):
         sync_repo(self.cfg, importer)
         repo = client.get(repo['_href'])
         self.assertNotEqual(latest_version_href, repo['_latest_version_href'])
+
+
+class SyncChangeRepoVersionTestCase(unittest.TestCase):
+    """Verify whether sync of repository updates repository version."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+
+    def test_all(self):
+        """Call :meth:`do_test` with varying arguments.
+
+        Call :meth:`do_test` with each possible pairing of
+        :data:`pulp_smash.tests.pulp3.constants.IMPORTER_DOWN_POLICY` and
+        :data:`pulp_smash.tests.pulp3.constants.IMPORTER_SYNC_MODE`. If `Pulp
+        #3320`_ affects Pulp, then the ``background`` and ``on_demand``
+        download policies are omitted from the test matrix.
+
+        .. _Pulp #3320: https://pulp.plan.io/issues/3320
+        """
+        importer_down_policy = IMPORTER_DOWN_POLICY
+        if selectors.bug_is_untestable(3320, self.cfg):
+            importer_down_policy -= {'background', 'on_demand'}
+        for pair in product(importer_down_policy, IMPORTER_SYNC_MODE):
+            with self.subTest(pair=pair):
+                self.do_test(*pair)
+
+    def do_test(self, download_policy, sync_mode):
+        """Verify whether the sync of a repository updates its version.
+
+        This test explores the design choice stated in the `Pulp #3308`_ that a
+        new repository version is created even if the sync does not add or
+        remove any content units. Even without any changes to the importer if a
+        new sync occurs, a new repository version is created.
+
+        .. _Pulp #3308: https://pulp.plan.io/issues/3308
+
+        Do the following:
+
+        1. Create a repository, and an importer.
+        2. Sync the repository an arbitrary number of times.
+        3. Verify that the repository version is equal to the previous number
+           of syncs.
+
+        :param download_policy: The download policy for the importer.
+        :param sync_mode: The sync mode for the importer.
+        """
+        client = api.Client(self.cfg, api.json_handler)
+        client.request_kwargs['auth'] = get_auth()
+        repo = client.post(REPO_PATH, gen_repo())
+        self.addCleanup(client.delete, repo['_href'])
+        body = gen_importer(repo)
+        body['download_policy'] = download_policy
+        body['feed_url'] = urljoin(FILE_FEED_URL, 'PULP_MANIFEST')
+        body['sync_mode'] = sync_mode
+        importer = client.post(FILE_IMPORTER_PATH, body)
+        self.addCleanup(client.delete, importer['_href'])
+
+        number_of_syncs = randint(1, 10)
+        for _ in range(number_of_syncs):
+            sync_repo(self.cfg, importer)
+
+        repo = client.get(repo['_href'])
+        path = urlsplit(repo['_latest_version_href']).path
+        latest_repo_version = path.split('/')[-2]
```

Comment:
Cast this string to an integer. Without a type cast, the `assertEqual` below is comparing a string to an integer. I'm sure the semantics of doing so are defined by `assertEqual`, but why delve into that territory?